libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/kpm/idea/IdeaKotlinProjectModelBuilder.kt

```diff
@@ -0,0 +1,24 @@
+package org.jetbrains.kotlin.gradle.kpm.idea
+
+import org.gradle.api.Project
+import org.gradle.tooling.provider.model.ToolingModelBuilder
+import org.jetbrains.kotlin.compilerRunner.konanHome
+import org.jetbrains.kotlin.gradle.dsl.pm20Extension
+import org.jetbrains.kotlin.gradle.plugin.mpp.pm20.KotlinPm20ProjectExtension
+import java.io.File
+
+class IdeaKotlinProjectModelBuilder : ToolingModelBuilder {
```

Comment:
If there is a use-case for building models for only some platforms (mostly to avoid resolving dependencies for fragments that you are not going to use), you can consider implementing `ParameterizedToolingModelBuilder` which allows injecting platform types that are trying to sync.

Reply:
Oooh! I really, really, really do like this idea! I see many projects already configuring targets based on `isIdeaActive` flags or other properties to avoid importing "unwanted" targets. This could later become an IDE feature that allows to select what platforms you really would like to import. 

I would, however, propose to delay this for a future PR (looking at this already growing past the 1k mark)